modules/top-level/output.nix

```diff
@@ -284,20 +302,24 @@ in
         '';
       };
 
+      # Set `wrapRc` and `impureRtp`s option defaults with even lower priority than `mkOptionDefault`
+      wrapRc = lib.mkOverride 1501 true;
+      impureRtp = lib.mkOverride 1501 false;
+
       extraConfigLuaPre = lib.mkOrder 100 (
-        # Add a global table at start of init
-        ''
-          -- Nixvim's internal module table
-          -- Can be used to share code throughout init.lua
-          local _M = {}
-        ''
-        + lib.optionalString config.wrapRc ''
-
-          -- Ignore the user lua configuration
-          vim.opt.runtimepath:remove(vim.fn.stdpath('config'))              -- ~/.config/nvim
-          vim.opt.runtimepath:remove(vim.fn.stdpath('config') .. "/after")  -- ~/.config/nvim/after
-          vim.opt.runtimepath:remove(vim.fn.stdpath('data') .. "/site")     -- ~/.local/share/nvim/site
-        ''
+        lib.concatStringsSep "\n" (
+          lib.optional (!config.impureRtp) ''
+            -- Ignore the user lua configuration
+            vim.opt.runtimepath:remove(vim.fn.stdpath('config'))              -- ~/.config/nvim
+            vim.opt.runtimepath:remove(vim.fn.stdpath('config') .. "/after")  -- ~/.config/nvim/after
+          ''
+          # Add a global table at start of init
+          ++ lib.singleton ''
```

Comment:
using singleton just for formatting here?

Reply:
Yeah, purely stylistic. I just felt it matched well with the use of `optional` above.

I've noticed a lot of nixpkgs code uses things like `singleton` to cut down on boilerplate and it felt suitable here.